test/filters/oidc/jwks_resolver_test.cc

```diff
@@ -155,6 +155,39 @@ TEST(JwksResolverTest, TestDynamicJwksResolverWithInvalidHttpStatus) {
   EXPECT_EQ(nullptr, resolver.jwks());
 }
 
+TEST(JwksResolverTest,
+     TestDynamicJwksResolver_RequestBy3SecInterval_UntilJwksConfigured) {
+  boost::asio::io_context io_context;
+  auto mock_http = std::make_shared<common::http::HttpMock>();
+
+  // Configured request interval as 10000 sec. This value is enough to guarantee
+  // that second request will not be invoked for 3 sec evloop run.
+  DynamicJwksResolverImpl resolver("istio.io", std::chrono::seconds(10000),
+                                   mock_http, io_context);
+
+  // Never initialized with invalid HTTP status.
+  EXPECT_CALL(*mock_http, Get(Eq("istio.io"), _, _, _, _, _, _))
+      .WillRepeatedly(
+          Invoke([](absl::string_view,
+                    const std::map<absl::string_view, absl::string_view>,
+                    absl::string_view, absl::string_view, absl::string_view,
+                    boost::asio::io_context&, boost::asio::yield_context) {
+            common::http::response_t response = std::make_unique<
+                beast::http::response<beast::http::string_body>>();
+            response->result(503);
+            return response;
+          }));
+
+  io_context.run_for(std::chrono::seconds(3));
```

Comment:
We should sleep slightly longer than interval right? This ensures the retries would be kicked in. say 4 seconds.